src/Doctrine/DoctrineHelper.php

```diff
@@ -85,43 +93,67 @@ public function getEntityNamespace(): string
     {
-        /** @var EntityManagerInterface $em */
-        $em = $this->getRegistry()->getManagerForClass($className);
+        try {
+            /** @var EntityManagerInterface $em */
+            $em = $this->getRegistry()->getManagerForClass($className);
+        } catch (\ReflectionException $exception) {
+            // this exception will be thrown by the registry if the class isn't created yet.
+            // an example case is the "make:entity" command, which needs to know which driver is used for the class to determine
+            // if the class should be generated with attributes or annotations. If this exception is thrown, we will check based on the
+            // namespaces for the given $className and compare it with the doctrine configuration to get the correct MappingDriver.
+
+            return $this->isInstanceOf($this->getMappingDriverForNamespace($className), $driverClass);
+        }
 
         if (null === $em) {
             throw new \InvalidArgumentException(sprintf('Cannot find the entity manager for class "%s"', $className));
         }
 
-        if (null === $this->annotatedPrefixes) {
+        if (null === $this->mappingDriversByPrefix) {
             // doctrine-bundle <= 2.2
             $metadataDriver = $em->getConfiguration()->getMetadataDriverImpl();
 
             if (!$this->isInstanceOf($metadataDriver, MappingDriverChain::class)) {
-                return $metadataDriver instanceof AnnotationDriver;
+                return $this->isInstanceOf($metadataDriver, $driverClass);
             }
 
             foreach ($metadataDriver->getDrivers() as $namespace => $driver) {
                 if (0 === strpos($className, $namespace)) {
-                    return $driver instanceof AnnotationDriver;
+                    return $this->isInstanceOf($driver, $driverClass);
                 }
             }
 
-            return $metadataDriver->getDefaultDriver() instanceof AnnotationDriver;
+            return $this->isInstanceOf($metadataDriver->getDefaultDriver(), $driverClass);
         }
 
         $managerName = array_search($em, $this->getRegistry()->getManagers(), true);
 
-        foreach ($this->annotatedPrefixes[$managerName] as [$prefix, $annotationDriver]) {
+        foreach ($this->mappingDriversByPrefix[$managerName] as [$prefix, $prefixDriver]) {
             if (0 === strpos($className, $prefix)) {
-                return null !== $annotationDriver;
+                return $this->isInstanceOf($prefixDriver, $driverClass);
             }
         }
 
         return false;
     }
 
+    public function isClassAnnotated(string $className): bool
+    {
+        return $this->doesClassUseDriver($className, AnnotationDriver::class);
+    }
+
+    public function doesClassUsesAttributes(string $className): bool
+    {
+        return $this->doesClassUseDriver($className, AttributeDriver::class);
+    }
+
+    public function canUseAttributes(): bool
```

Comment:
This looks identical in purpose to `isDoctrineSupportingAttributes()`, but this method looks incorrect and `isDoctrineSupportingAttributes` looks correct. Unless I'm missing something, we should use `isDoctrineSupportingAttributes()` everywhere and remove this method.

Reply:
done